### ISSUE TITLE
VSCode Remote SSH Extention

### ISSUE THREAD

**Issue by sharksforarms** [onivim/oni2#2277]
Hello, I was able to install the VSCode Remote SSH extension (remote dev work over ssh)

https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh

Although it doesn't seem like I'm able to use it. Options appear in the Ctrl+Shift+P menu, but selecting them does nothing

**Comment by CrossR** [comment]
Hi!

We haven't got support for every extension right now, and the ones that are working are the ones that add language features. That will in time extend to include some out of editor stuff like debugger support, output from running things, tree-features (custom file-explorer like trees for code structure say) etc.

However, for that specific extension, we can't support it as it is not open source. You can find the license on the link you've posted and part of that license is:

> You may use a copy of the software with each validly licensed copy of Microsoft Visual Studio Code. You may not use the software if you do not have a license for Microsoft Visual Studio Code.

That, and the fact that I believe the actual implementation isn't open source so we couldn't implement the Oni2 side of things either (not 100% on that one actually), mean we can't support that extension. There is a few extensions which aren't possible due to their licensing, including the similar Docker container extension.

If at some point in the future that license changes, we will do our best to support it since so many people have asked!

**Comment by sharksforarms** [comment]
Makes sense thanks! Would be great to have a "remote development" story for oni

**Comment by geohuz** [comment]
I've tried workaround of installing sshfs to enable remote editing, it worked except using``oni2 .`` then the file explorer on the left side is empty. It worked for entering individual file name though.

**Comment by gordonrust** [comment]
A highly recommended vaiable alternative to the vscode plugin https://mutagen.io/

I have been using this for 2 months now.
Though one needs to have the source files on the local and this tools essentialy does an intelligent syncing , I have found this 
to really obviate the need for vscode. I have actually switched back to nvim :).

**Comment by JacobSandin** [comment]
Is Remote SSH still in the distant future?
